server/src/main/java/org/elasticsearch/cluster/metadata/MetadataUpdateSettingsService.java

```diff
@@ -443,6 +453,17 @@ private static boolean maybeUpdateClusterBlock(
                         changed = true;
                         if (block.contains(ClusterBlockLevel.WRITE)) {
```

Comment:
In relation to my comment on the v9 PR, just noting that this should follow whatever we do on that PR.

Reply:
I updated the behavior in the V9 PR and here, see this [comment](https://github.com/elastic/elasticsearch/pull/120648#issuecomment-2615315991)